modules/opta-k8s-service-helm/templates/crons.yaml

```diff
@@ -0,0 +1,62 @@
+{{- range $index, $cron := .Values.crons }}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "k8s-service.fullname" $ }}-{{ $index }}
+  labels:
+    tags.datadoghq.com/service: {{ include "k8s-service.serviceName" $ }}-{{ include "k8s-service.namespaceName" $ }}
+    tags.datadoghq.com/version: {{ $.Values.version | quote }}
+  {{- include "k8s-service.labels" $ | nindent 4 }}
+  namespace: {{ include "k8s-service.namespaceName" $ }}
+spec:
+  schedule: {{ $cron.schedule | quote }}
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: "Never"
+          {{- if ne ( len $.Values.tolerations ) 0 }}
+          tolerations:
+          {{ range $toleration :=  $.Values.tolerations }}
+          - key: {{ $toleration.key | quote }}
+            value: {{ default "opta" $toleration.value | quote }}
+            effect: {{ default "NoSchedule" $toleration.effect | quote }}
+          {{ end }}
+          {{- end }}
+          containers:
+            - name: cron
+              image: "{{ $.Values.image }}"
```

Comment:
So if we have a "cron" in the opta service file in a k8s-service module, then we are installing a "deployment" (the current behavior) and this "cronjob" resource into Kubernetes with the identical image and environment in both the deployment and the cronjob. That should be useful in e.g. running migrations or backups etc. periodically. 

But...then we have wedded the two aspects; so when one changes, the other does too. Should be fine but its a little unconventional to combine them. Although I cant think of another way given that the k8s-service deploy command accepts a single image name/tag only so for example we dont have the possibility of creating two different k8s-service modules, one for deployment and one for cronjob.  

I'd rather use the k8s-resource and define a separate cronjob there, but thats just a choice.

Reply:
Great insight, 2 points:
1. So we can totally just have a pure cronjob service by specifying the min/max containers to be 0-- then only the cronjobs would run!
2. The special treatment we've given to the image has bitten us so much in the past, but it looks from today's discussion and plans for the onsite that we'll be trying to purge it very soon

Reply:
Ok

Reply:
done